diff --git a/pyzbar/zbar_library.py b/pyzbar/zbar_library.py
index b63108c..bf4a060 100644
--- a/pyzbar/zbar_library.py
+++ b/pyzbar/zbar_library.py
@@ -60,10 +60,7 @@ def load():
             dependencies, libzbar = load_objects(Path(__file__).parent)
     else:
         # Assume a shared library on the path
-        path = find_library('zbar')
-        if not path:
-            raise ImportError('Unable to find zbar shared library')
-        libzbar = cdll.LoadLibrary(path)
+        libzbar = cdll.LoadLibrary("libzbar.so.0")
         dependencies = []
 
     return libzbar, dependencies